src/ckeditor.js

```diff
@@ -12,9 +14,56 @@
  * @singleton
  */
 
-CKEDITOR.define( function() {
+CKEDITOR.define( [ 'editor', 'mvc/collection', 'promise' ], function( Editor, Collection, Promise ) {
 	var CKEDITOR = {
 		/**
+		 * A collection containing all editor instances created.
+		 */
+		instances: new Collection(),
+
+		/**
+		 * Creates an editor instance for the provided DOM element.
+		 *
+		 * The creation of editor instances is an asynchronous operation, therefore a promise is returned by this
+		 * method.
+		 *
+		 *     CKEDITOR.create( '#content' );
+		 *
+		 *     CKEDITOR.create( '#content' ).then( function( editor ) {
+		 *         // Manipulate "editor" here.
+		 *     } );
+		 *
+		 * @param {String|Object} element An element selector or a DOM element, which will be the source for the
+		 * created instance.
+		 * @returns {Promise} A promise, which will be fulfilled with the created editor.
+		 */
+		create: function( element ) {
+			var that = this;
+
+			return new Promise( function( resolve, reject ) {
+				// If a query selector has been passed, transform it into a real element.
+				if ( typeof element == 'string' ) {
+					element = document.querySelector( element );
```

Comment:
Ofc consistency is most very important here, therefore I consider only 2 and 3 as viable strategies.

> 2. Always pass an array back, even for one editor only. Empty array if no element found.

I like this option more.

* If I'm very lazy developer I'll simply put `CKEDITOR.create('.editor')` on each page. I don't care if on given subpage there's one editor or none.
* Selectors might be very generic, it's a perfectly normal situation that it they won't match any element on a page.
	* `document.querySelectorAll` does not consider it to be erorr if our selector does not match anything.
	* jQuery does not either.

That makes me think that it's fine that no editors were created with my selector. It won't do any harm, because in `then()` method I'm `for each`ing result anyway because we work on array.

Verdict: option 2